src/compiler/transformers/es6.ts

```diff
@@ -834,12 +897,38 @@ namespace ts {
             return block;
         }
 
-        function transformConstructorBodyWithSynthesizedSuper(node: ConstructorDeclaration) {
-            return visitNodes(node.body.statements, visitor, isStatement, 1);
+        function makeTransformerForConstructorBodyAtOffset(offset: number): (c: ConstructorDeclaration) => NodeArray<Statement> {

```

Comment:
Is there a reason for this function rather than just putting the closure in `transformConstructorBody` directly?